From: Robert Luberda <robert@debian.org>
Date: Sun, 14 Feb 2016 11:37:10 +0100
Subject: Use endian.h to detect byte ordering

Include <endian.h> and check value of the __BYTE_ORDER
macro to properly detect used byte ordering. This should
fix FTBFS on archs like alpha, hppa or sparc64.
---
 C/CpuArch.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/C/CpuArch.h b/C/CpuArch.h
index 69478a3..74896a6 100644
--- a/C/CpuArch.h
+++ b/C/CpuArch.h
@@ -5,6 +5,7 @@
 #define __CPU_ARCH_H
 
 #include "7zTypes.h"
+#include <endian.h>
 
 EXTERN_C_BEGIN
 
@@ -60,6 +61,7 @@ If MY_CPU_LE_UNALIGN is not defined, we don't know about these properties of pla
     || defined(MY_CPU_ARM_LE) \
     || defined(MY_CPU_IA64_LE) \
     || defined(__LITTLE_ENDIAN__) \
+    || __BYTE_ORDER == __LITTLE_ENDIAN \
     || defined(__ARMEL__) \
     || defined(__THUMBEL__) \
     || defined(__AARCH64EL__) \
@@ -70,6 +72,7 @@ If MY_CPU_LE_UNALIGN is not defined, we don't know about these properties of pla
 #endif
 
 #if defined(__BIG_ENDIAN__) \
+    || __BYTE_ORDER == __BIG_ENDIAN \
     || defined(__ARMEB__) \
     || defined(__THUMBEB__) \
     || defined(__AARCH64EB__) \
